### PR TITLE
Display standalone parameter names by default

### PR DESCRIPTION
**Pull request by jjonescz**
Closes https://github.com/dotnet/roslyn/issues/67478.
Closes https://github.com/dotnet/roslyn/issues/67632.
Closes https://github.com/dotnet/roslyn/issues/67464.
Motivated by https://github.com/dotnet/roslyn/issues/67478#issuecomment-1485321129.

Changes symbol display formatter to include names if used on `IParameterSymbol` directly by default (unless compiler-internal option is specified).

### CODE REVIEW

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4)

**Comment by jjonescz** [comment]
@dotnet/roslyn-compiler for a second review
@CyrusNajmabadi FYI

**Comment by jcouv** [comment]
https://github.com/orgs/dotnet/teams/roslyn-compiler for a second review

File: docs/Breaking API Changes.md
Comment (AlekseyTs) [inline docs/Breaking API Changes.md:80]: What is this version for?

**Comment by AlekseyTs** [comment]
Did we get an official approval for the breaking change? Otherwise, commit 5 looks good, but I am not signing off yet.

File: docs/Breaking API Changes.md
Comment (CyrusNajmabadi) [inline docs/Breaking API Changes.md:78]: FYI, we called these 'Display' apis intentionally so that we coudl change them without it being considered a breaking change.  I'm fine documenting this though, just wanted to reiterate our position that things that exist for 'display purposes' absolutely can and will change depending on what we think is best for displaying them :)

File: docs/Breaking API Changes.md
Comment (AlekseyTs) [inline docs/Breaking API Changes.md:86]: It looks like this section is completely unrelated to the goal of this PR

Comment (AlekseyTs) [reply docs/Breaking API Changes.md:86]: > It looks like this section is completely unrelated to the goal of this PR

I guess it came from the parent branch with the merge.

Comment (jjonescz) [reply docs/Breaking API Changes.md:86]: Yes, this PR hasn't touched that section, though.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 7)

**Pull request merged**
